Gems/Prefab/PrefabDependencyViewer/Code/Tests/PrefabDependencyTreeGenerationTests.cpp

```diff
@@ -0,0 +1,234 @@
+
+    bool str_equal(const char* expected, const char* result)
+    {
+        return (AZStd::string)expected == (AZStd::string)result;
+    }
+
+    void EXPECT_STR_EQ(const char* expected, const char* result)
+    {
+        EXPECT_EQ(true, str_equal(expected, result));
+    }
+
+    TEST_F(PrefabDependencyViewerFixture, INVALID_TEMPLATE_ID)
+    {
+        Outcome outcome = PrefabDependencyTree::GenerateTreeAndSetRoot(
+                            InvalidTemplateId, m_prefabSystemComponent);
+
+        EXPECT_EQ(false, outcome.IsSuccess());
+    }
+
+    TEST_F(PrefabDependencyViewerFixture, EMPTY_PREFAB_NO_SOURCE_TEST)
+    {
+        TemplateId tid = 10;
+        EXPECT_CALL(*m_prefabSystemComponent, FindTemplateDom(tid))
+            .Times(1)
+            .WillRepeatedly(::testing::ReturnRef(m_prefabDomsCases["emptyJSON"]));
+
+        Outcome outcome = PrefabDependencyTree::GenerateTreeAndSetRoot(10, m_prefabSystemComponent);
+        EXPECT_EQ(false, outcome.IsSuccess());
+    }
+
+    TEST_F(PrefabDependencyViewerFixture, EMPTY_PREFAB_WITH_SOURCE_TEST)
+    {
+        TemplateId tid = 2000;
+        EXPECT_CALL(*m_prefabSystemComponent, FindTemplateDom(tid))
+            .Times(1)
+            .WillRepeatedly(::testing::ReturnRef(m_prefabDomsCases["emptyJSONWithSource"]));
+
+        Outcome outcome = PrefabDependencyTree::GenerateTreeAndSetRoot(tid, m_prefabSystemComponent);
+        EXPECT_EQ(true, outcome.IsSuccess());
+
+        PrefabDependencyTree tree = outcome.GetValue();
+        EXPECT_EQ(tid, tree.GetRoot()->GetMetaData().GetTemplateId());
+        EXPECT_STR_EQ("Prefabs/emptySavedJSON.prefab", tree.GetRoot()->GetMetaData().GetSource());
+
+        EXPECT_EQ(0, tree.GetChildren(tree.GetRoot()).size());
+    }
+
+    TEST_F(PrefabDependencyViewerFixture, NESTED_PREFAB_WITH_ATLEAST_ONE_INVALID_SOURCE_FILE)
+    {
+        TemplateId tid = 52893;
+        EXPECT_CALL(*m_prefabSystemComponent, FindTemplateDom(tid))
+            .Times(1)
+            .WillRepeatedly(::testing::ReturnRef(m_prefabDomsCases["NestedPrefabWithAtleastOneInvalidNestedInstance"]));
+
+        EXPECT_CALL(*m_prefabSystemComponent, GetTemplateIdFromFilePath(AZ::IO::PathView("Prefabs/goodPrefab.prefab")))
+            .Times(1)
+            .WillRepeatedly(::testing::Return(5));
+
+        EXPECT_CALL(*m_prefabSystemComponent, GetTemplateIdFromFilePath(AZ::IO::PathView("")))
+            .Times(1)
+            .WillRepeatedly(::testing::Return(InvalidTemplateId));
+
+        // Depending on which TemplateId stack gets to first
+        // you can or can't call FindTemplateDom for GoodNestedPrefab's
+        // TemplateId
+        EXPECT_CALL(*m_prefabSystemComponent, FindTemplateDom(5))
+            .Times(0);
+
+        Outcome outcome = PrefabDependencyTree::GenerateTreeAndSetRoot(tid, m_prefabSystemComponent);
+        EXPECT_EQ(false, outcome.IsSuccess());
+    }
+
+    NodeList FindNodes(Utils::NodeSet& nodeSet, TemplateId tid, const char* source)
```

Comment:
Functions that help with testing typically go into the test fixture.

Reply:
Do they have to specifically go inside the class or struct Fixture or just within the file?

Reply:
The fixture class.

Reply:
Resolved